Sources/Mantle/Check.swift

```diff
@@ -455,7 +455,12 @@ extension TypeChecker where PhaseState == CheckPhaseState {
           fatalError()
         }
         return (.absurd, type)
-      case let .constructor(dataCon, synPats):
+      case let .constructor(dataConSet, synPats):
+        guard
+          let dataCon = self.disambiguateConstructor(dataConSet, patType)
+        else {
+          fatalError()
```

Comment:
Message, please.

Reply:
Can't without refactoring.  This routine "can't fail" given its current type.